k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -173,11 +173,23 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
         case Some((event, ws)) =>
           import Ordering.Implicits._
           // Register the update only if its resource version is larger than the largest version
-          // seen so far.
+          // seen so far, or the incoming event is a DELETE event.
           if (largestEvent.forall(_ < event)) {
+            log.trace(
+              "k8s watch on '%s' registered event with newer resource version %s (older resource version was %s)",
+              watchPath,
+              event.resourceVersion,
+              largestEvent.flatMap(_.resourceVersion)
+            )
             state.update(Activity.Ok(event))
             _processEventStream(ws, event.resourceVersion, Some(event))
           } else {
+            log.trace(
```

Comment:
Friendly reminder about potentially making this a `log.warn`  (or `info`?), assuming this PR will supercede #1810. :)

Reply:
👍

Reply:
👍